examples/benchmarking.ipynb

```diff
@@ -0,0 +1,180 @@
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Next, let's initialize our Battery Circuit by creating model object."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Step 1: Create a model object\n",
+    "batt = BatteryCircuit()"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Then, for our model, we would need to define a future loading function. More information on what a future loading function is and how to use it can be found here: https://nasa.github.io/progpy/prog_models_guide.html#future-loading"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Step 2: Define future loading function \n",
+    "loading = batt.InputContainer({'i': 2})  # Constant loading\n",
+    "def future_loading(t, x=None):\n",
+    "    # Constant Loading\n",
+    "    return loading"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Since this is a simple example, we are going to have a constant loading!"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Finally, we are going to need to benchmark the simulation!\n",
+    "\n",
+    "We can do this by using the `timeit()` function and pass in our `simulate_to()` or `simulate_to_threshold()` function for the `stmt` argument. For more information regarding the `timeit()` function, please read it's documentation located here: https://docs.python.org/3/library/timeit.html"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Step 3: Benchmark simulation of 600 seconds\n",
+    "def sim():  \n",
+    "    batt.simulate_to(600, future_loading)\n",
+    "time = timeit(sim, number=500)"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In this example, we would be benchmarking the simulation for the BatteryCircuit model up to 600 seconds. Furthermore, we will define our `number` argument to be 500 for sake of runtime."
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Let's print out the results of the benchmark test!"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Print results\n",
+    "print('Simulation Time: {} ms/sim'.format(time))"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In this example, we benchmarked the simulation of the BatteryCircuit model up to 600 seconds. By utilizing the `time` package, we are able to benchmark the simulation of the model and print out the results with ease!"
```

Comment:
These two sentences feel a tad repetitive. Maybe we could adjust one to have a description of what benchmarking means in this context? Open to your suggestions.

Reply:
Yeah, I agree, how does this sound:

'In this example, we benchmarked the simulation of the BatteryCircuit model up to 600 seconds by utilizing the `time` package!'

Reply:
Sounds good, nice job.